File: ChristmasTree/arvore.py
tamanho = 12    #Define o tamanho da árvore em altura

def construir_arvore(tamanho):  #Marca o inicio da função e usar 'tamanho' como propriedade
    largura_base = tamanho * 2 - 1  #Define o tamanho da base de acordo com a altura da arvore

    print(" " * (tamanho - 1) + "*")    #Imprime a estrela no ponto mais alto da árvore

    for i in range(tamanho):    #Cria um laço de repetição para fazer a copa da árvore 
        folhas = "=" * (2 * i + 1)  #Define quantas folhas vão ter na linha
        espacos = " " * (tamanho - i - 1)   #Define quantos espações vão ter na linha
        print(espacos + folhas - espacos)   #Imprime os espaços e as folhas

    tronco = " " * (tamanho - 1) + "|" + " " * (tamanho - 1)    #Imprime o tronco da árvore

construir_arvore(tamanho)   #Chama a função

File: ChristmasTree/arvore.js
const tamanho = 12; // Você pode mudar esse valor para árvores maiores

const construirArvore = (tamanho) => {
  const larguraBase = tamanho * 2 - 1;

  // Estrela no topo
  console.log(" ".repeat(tamanho - 1) + "*");

  // Corpo da árvore
  for (let i = 0; i < tamanho; i++) {
    const folhas = "=".repeat(2 * i + 1);
    const espacos = " ".repeat(tamanho - i - 1);
    console.log(espacos + folhas + espacos);
  }

  // Tronco
  const tronco = "_".repeat(tamanho - 1) + "|" + "_".repeat(tamanho - 1);
  console.log(tronco);

};

construirArvore(tamanho);
